docker/loadbalancer.go

```diff
@@ -0,0 +1,136 @@
+	"sigs.k8s.io/cluster-api-provider-docker/third_party/forked/loadbalancer"
+	clusterapiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/kind/pkg/cluster/constants"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+)
+
+// LoadBalancer implement a service for managing the docker containers hosting a loadbalancer.
+type LoadBalancer struct {
+	logr.Logger
+	Cluster       *clusterapiv1alpha2.Cluster
+	DockerCluster *infrav1.DockerCluster
+}
+
+// NewLoadBalancer returns a new LoadBalancer service for the given Cluster/DockerCluster pair.
+func NewLoadBalancer(cluster *clusterapiv1alpha2.Cluster, dockerCluster *infrav1.DockerCluster, logger logr.Logger) (*LoadBalancer, error) {
+	if cluster == nil {
+		return nil, errors.New("failed to generate new scope from nil Cluster")
+	}
+	if dockerCluster == nil {
+		return nil, errors.New("failed to generate new scope from nil DockerCluster")
+	}
+
+	if logger == nil {
+		logger = klogr.New()
+	}
+
+	return &LoadBalancer{
+		Cluster:       cluster,
+		DockerCluster: dockerCluster,
+		Logger:        logger,
+	}, nil
+}
+
+// Create a docker containers hosting a loadbalancer for the cluster.
+func (s *LoadBalancer) Create() (*nodes.Node, error) {
+	// Describe and create if not exists.
+	lb, err := s.container()
+	if err != nil {
+		return nil, err
+	}
+
+	if lb == nil {
+		s.V(4).Info("Creating load balancer container")
+		lb, err = nodes.CreateExternalLoadBalancerNode(
+			s.containerName(),
+			loadbalancer.Image,
+			s.ClusterLabel(),
+			"0.0.0.0",
+			0,
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return lb, nil
+}
+
+// Delete the docker containers hosting a loadbalancer for the cluster.
+func (s *LoadBalancer) Delete() error {
+	// Describe and delete if exists.
+	lb, err := s.container()
+	if err != nil {
+		return err
+	}
+
+	if lb != nil {
+		s.V(4).Info("Deleting load balancer container")
+		if err := nodes.Delete(*lb); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// containerName is the name of the docker container with the load balancer
+func (s *LoadBalancer) containerName() string {
+	return fmt.Sprintf("%s-lb", s.Cluster.Name)
+}
+
+// ClusterLabel is the label of the docker containers for this cluster
+func (s *LoadBalancer) ClusterLabel() string {
+	return fmt.Sprintf("%s=%s", constants.ClusterLabelKey, s.Cluster.Name)
+}
+
+// roleLabel is the label of the docker container with the load balancer
+func (s *LoadBalancer) roleLabel() string {
+	return fmt.Sprintf("%s=%s", constants.NodeRoleKey, constants.ExternalLoadBalancerNodeRoleValue)
+}
+
+// container returns the docker container with the load balancer
+func (s *LoadBalancer) container() (*nodes.Node, error) {
+	n, err := nodes.List(
+		fmt.Sprintf("label=%s", s.ClusterLabel()),
+		fmt.Sprintf("label=%s", s.roleLabel()),
+		fmt.Sprintf("name=%s", s.containerName()),
+	)
+	if err != nil {
+		return nil, errors.Wrapf(errors.WithStack(err), "Failed to list load balancer containers")
```

Comment:
`errors.Wrap` will wrap the err with a stack trace, there is no need for the `WithStack` call.

errors should all start with lowercase letters, please see https://github.com/golang/go/wiki/CodeReviewComments#error-strings